### ISSUE TITLE
Nested viewports don't work correctly with get_local_mouse_position()

### ISSUE THREAD

**Issue by ocdmonkey** [godotengine/godot#19685]
**Godot version:**
3.0.2

**OS/device including version:**
Windows 10

**Issue description:**
When nesting viewports (viewportcontainer/viewport/viewportcontainer/viewport), the mouse input is always scaled from the root viewport instead of the parent.

**Steps to reproduce:**
Make 4 nodes with the following hierarchy:
viewportcontainer/viewport/viewportcontainer2/viewport2
scale viewportcontainer, and call get_local_mouse_position() on viewportcontainer2. It will pass the local position as if it was not a child of viewportcontainer.

**Comment by akien-mga** [comment]
Is this still reproducible in the current master branch / recent 3.1 builds?

**Comment by ocdmonkey** [comment]
Yes, this is still a problem. I've done some more testing and compiled it all into an example project. Basically, the only way to get the correctly scaled mouse position is to call get_local_mouse_position() from the viewportcontainer you scaled. Calling it from any children nodes will return a value as if there wasn't any scaling.

[viewport glitch test.zip](https://github.com/godotengine/godot/files/3806418/viewport.glitch.test.zip)

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?
(I tried the reproduction project, but not sure what is the expected result)

**Comment by CowThing** [comment]
This issue is still present in 3.3 stable. Both `get_local_mouse_position()` and `get_global_mouse_position()` give incorrect results in a scaled viewport. From what I can tell the issue is that these functions are calling `get_viewport().get_mouse_position()`
https://github.com/godotengine/godot/blob/476bc5191bafcdea99e918100394479c06f0e5b4/scene/2d/canvas_item.cpp#L1086

Which is getting the mouse position directly from the Input singleton. Which I guess is not transformed correctly.
https://github.com/godotengine/godot/blob/476bc5191bafcdea99e918100394479c06f0e5b4/scene/main/viewport.cpp#L1511

To fix this the mouse position should be gotten from the `_input()` function, which is correctly transformed. For example here's how to fix the problem in the earlier test project

```GDScript
extends Node2D

var last_mouse_position : = Vector2()

func _input(event : InputEvent) -> void:
	if event is InputEventMouseMotion:
		last_mouse_position = event.position

func get_correct_global_mouse_position() -> Vector2:
	return get_canvas_transform().affine_inverse().xform(last_mouse_position)

func get_correct_local_mouse_position() -> Vector2:
	return get_global_transform().affine_inverse().xform(get_correct_global_mouse_position())


func _process(delta: float) -> void:
	$icon.position = get_correct_local_mouse_position()
```
Hopefully a fix like this can be translated to C++, because there are some nodes that use `get_local_mouse_position()` internally.

By the way this issue doesn't only happen with nested viewports, just a single scaled viewport will have this issue.

**Comment by robfram** [comment]
Using the insightful analysis shared by @CowThing, I tried to replace the original implementation with that behaviour in the C++ Viewport class.

I modified the project by @ocdmonkey to test different cases, using also the GDScript fix to test against.
[viewport-glitch.zip](https://github.com/godotengine/godot/files/7477274/viewport-glitch.zip)
